staging/kos/pkg/controllers/subnet/validator.go

```diff
@@ -228,33 +227,33 @@ func (v *Validator) processDeletedSubnet(s k8stypes.NamespacedName) {
 func (v *Validator) processExistingSubnet(s *netv1a1.Subnet) error {
 	ss, parsingErrs := subnet.NewSummary(s)
 	if len(parsingErrs) > 0 {
-		return parsingErrs
+		glog.Errorf("Subnet %s/%s is malformed: %s. This should never happen.", s.Namespace, s.Name, parsingErrs.Error())
+		return nil
 	}
 
 	if v.subnetIsStale(ss.NamespacedName, s.ResourceVersion) {
-		return nil
+		return fmt.Errorf("current version (%s) is stale", s.ResourceVersion)
 	}
 
-	// If we're here s might have been created or updated in a way that affects
-	// validation. We need to update its conflicts cache accordingly and
-	// reconsider old rivals because they might no longer be in conflict with
-	// s.
-	oldRivals := v.updateConflictsCache(ss)
+	// Initialize s's conflicts cache if it's the first time s is processed. In
+	// case the conflicts cache still records a deleted subnet with same
+	// namespaced name as s return old rivals and reconsider them: the conflict
+	// might have disappeared.
+	oldRivals := v.initConflictsCache(ss.NamespacedName, ss.UID)
```

Comment:
The lead sentence of the comment, and the function name, have been changed to refer only to the init case.  But that is not the only case.

Reply:
Rolled back function name and tweaked comment in 7690042.
The rationale for the name change was that since updates to a subnet are forbidden when that function is executed and a new conflicts cache must be added it is always being initialized for a new subnet. When we had updates, a new conflicts cache could be added (after deleting the stale one)  because of an update to the same subnet. But practically speaking there's no difference: we still have the removal of an old conflicts cache and the addition of a new one, which is equivalent to an update, so I reverted the change.